hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -1139,6 +1158,30 @@ bool XmlChangeset::matchesChangesetPreconditionFailure(const QString& hint,
   return false;
 }
 
+bool XmlChangeset::matchesChangesetConflictVersionMismatchFailure(const QString& hint,
+                                                                  long& element_id, ElementType::Type& element_type,
+                                                                  long& version_old, long& version_new)
+{
+  //  Changeset conflict: Version mismatch: Provided 2, server had: 1 of Node 4869875616
+  QRegularExpression reg(
+        "Changeset conflict: Version mismatch: Provided ([0-9]+), server had: ([0-9]+) of (Node|Way|Relation) ([0-9]+)",
+        QRegularExpression::CaseInsensitiveOption);
+  QRegularExpressionMatch match = reg.match(hint);
+  if (match.hasMatch())
+  {
+    bool success = false;
+    version_old = match.captured(1).toLong(&success);
+    if (!success)
+      return success;
+    version_new = match.captured(2).toLong(&success);
+    if (!success)
+      return success;
+    element_type = ElementType::fromString(match.captured(3).toLower());
+    element_id = match.captured(4).toLong(&success);
+    return success;
+  }
+  return false;
+}
```

Comment:
This is the function that parses the error response from the OSM API server for mismatched versions, getting the element type, id, and both versions so that it can be updated.